src/duplex.ts

```diff
@@ -185,19 +186,22 @@ function _generate(mirror, obj, patches, path) {
       var newVal = obj[key];
 
       if (typeof oldVal == "object" && oldVal != null && typeof newVal == "object" && newVal != null) {
-        _generate(oldVal, newVal, patches, path + "/" + escapePathComponent(key));
+        _generate(oldVal, newVal, patches, path + "/" + escapePathComponent(key), invertible);
       }
       else {
         if (oldVal !== newVal) {
           changed = true;
+          if (invertible) patches.push({ op: "test", path: path + "/" + escapePathComponent(key), value: _deepClone(oldVal) });
```

Comment:
Could you add `{}` for `if`s. In my opinion it would make the code more readable.